### PR TITLE
Some update for Fedora supporting

### PR DESCRIPTION
**Pull request by wangli5665**
Some update for runltp-ng:
1. add support for fedora
2. support excluding more tests (e.g. --exclude="access03 access04")
3. support including more tests (e.g. --include="access02 access03")
4. use SPDX-License-Identifier
5. rebase to the latest main branch

### CODE REVIEW

**Comment by pevik** [comment]
`Reviewed-by: Petr Vorel <petr.vorel@gmail.com>` to whole patchset. although it's simple enough to be merged, I prefer to wait for @metan-ucw.

File: install_pkg.pm [hunk 1]
```diff
@@ -46,6 +32,12 @@ sub foo_to_pkg
 		'libattr-devel-suse' => 'libattr-devel',
 		'libcap-devel-suse' => 'libcap-devel',
 		'libnuma-devel-suse' => 'libnuma-devel',
+
+		'libaio-devel-fedora' => 'libaio-devel',
+		'libacl-devel-fedora' => 'libacl-devel',
+		'libattr-devel-fedora' => 'libattr-devel',
+		'libcap-devel-fedora' => 'libcap-devel',
+		'libnuma-devel-fedora' => 'numactl-devel',
```
Comment (pevik) [inline install_pkg.pm:40]: @wangli5665 nit: maybe sort distros alphabetically (I'm planning to add at least Alpine).

File: install_pkg.pm [hunk 2]
```diff
@@ -108,11 +103,13 @@ sub install_pkg
 
 	} elsif ($distro eq 'suse') {
 		return 'zypper --non-interactive --ignore-unknown in ' . join(' ', @pkgs);
+
+	} elsif ($distro eq 'fedora') {
+		return 'yum install -y ' . join(' ', @pkgs);
```
Comment (pevik) [inline install_pkg.pm:108]: @wangli5665 and here also sort alphabetically

**Review by pevik — APPROVED** [review]

File: install_pkg.pm (hunk 1 above)
Comment (wangli5665) [reply install_pkg.pm:40]: Agreed  +1

**Comment by metan-ucw** [comment]
The whole patchset looks good to me.

I do have local changes that add a few more dependencies into install phase, mkfs, quota, etc.
```
diff --git a/install_pkg.pm b/install_pkg.pm
index 1ea2516..dfb300b 100755
--- a/install_pkg.pm
+++ b/install_pkg.pm
@@ -34,7 +34,27 @@ sub foo_to_pkg
                'pkg-config' => 'pkg-config',
                'make' => 'make',
                'gcc' => 'gcc',
+               'bc' => 'bc',
 
+               # mkfs.foo
+               'dosfstools' => 'dosfstools',
+               'xfsprogs' => 'xfsprogs',
+               'e2fsprogs' => 'e2fsprogs',
+               'btrfsprogs' => 'btrfsprogs',
+               'btrfsprogs-debian' => 'btrfs-progs',
+
+               # FS quota tools
+               'quota' => 'quota',
+
+               # NFS tools
+               'nfs-utils' => 'nfs-utils',
+               'nfs-utils-debian' => 'nfs-kernel-server',
+
+               # kernel devel
+               'kernel-devel' => 'kernel-devel',
+               'kernel-devel-debian' => 'linux-headers-`uname -r`',
+
+               # devel libs
                'libaio-devel-debian' => 'libaio-dev',
                'libacl-devel-debian' => 'libacl1-dev',
                'libattr-devel-debian' => 'libattr1-dev',
@@ -136,7 +156,7 @@ sub install_ltp_pkgs
        return unless defined($distro);
 
        # Attempt to install required packages
-       my @required_pkgs = ('make', 'autoconf', 'automake', 'pkg-config', 'gcc');
+       my @required_pkgs = ('make', 'autoconf', 'automake', 'pkg-config', 'gcc', 'bc');
 
        # We need at least one
        push(@required_pkgs, 'git');
@@ -151,6 +171,24 @@ sub install_ltp_pkgs
                'libnuma-devel');
        push(@required_pkgs, @devel_libs);
 
+       # We need mkfs.foo at runtime
+       my @mkfs = (
+               'dosfstools',
+               'xfsprogs',
+               'e2fsprogs',
+               'btrfsprogs',
+       );
+       push(@required_pkgs, @mkfs);
+
+       # FS quota tests needs quota tools
+       push(@required_pkgs, ('quota'));
+
+       # NFS tests needs exportfs
+       push(@required_pkgs, ('nfs-utils'));
+
+       # Kernel devel so that we can build modules
+       push(@required_pkgs, ('kernel-devel'));
+
        my @cmds = ();
        push(@cmds, update_pkg_db($distro));
        push(@cmds, install_pkg($distro, \@required_pkgs));
```

It may be cleaner to apply these changes first then update the patch that adds fedora support so that we have all required fedora packages added in a single patch.

**Comment by wangli5665** [comment]
> It may be cleaner to apply these changes first then update the patch that adds fedora support so that we have all required fedora packages added in a single patch.

Sure, feel free to update your main branch, and I will rebase the patch accordingly.

**Comment by pevik** [comment]
And I then update https://github.com/metan-ucw/runltp-ng/pull/30.

**Comment by metan-ucw** [comment]
FYI I've pushed the changes.

**Comment by wangli5665** [comment]
FYI rebased to the latest.

**Comment by pevik** [comment]
@wangli5665 maybe you want to add Fedora setup in `pkg_to_m32()`.

**Comment by wangli5665** [comment]
> @wangli5665 maybe you want to add Fedora setup in `pkg_to_m32()`.

Sure, I'll have a look when available. Thanks!

**Pull request closed without merging**